### PR TITLE
infra: use image proxy p2

### PR DESCRIPTION
**Pull request by idoshamun**
Add the missing bits:
* boot
* comments
* post content
* uploaded images
* top reader
* fallback image

### CODE REVIEW

File: src/common/cloudinary.ts [hunk 1]
```diff
@@ -108,8 +108,12 @@ export const uploadPostFile = (
   preset: PostPreset,
 ) => uploadFile(name, preset, stream);
 
-export const mapCloudinaryUrl = (url?: string): string | undefined =>
-  url?.replace(
+export function mapCloudinaryUrl(url: string): string;
+export function mapCloudinaryUrl(url: undefined): undefined;
+export function mapCloudinaryUrl(url?: string): string | undefined;
+export function mapCloudinaryUrl(url?: string): string | undefined {
```
Comment (rebelchris) [inline src/common/cloudinary.ts:114]: Is this intentional/correct?
If so why is it needed?

**Review by rebelchris — CHANGES_REQUESTED** [review]
One thing that seems off.

File: src/common/cloudinary.ts (hunk 1 above)
Comment (rebelchris) [reply src/common/cloudinary.ts:114]: Surely you can do the last one alone and early return if no value is given?
If you accept string|undefined?

Comment (idoshamun) [reply src/common/cloudinary.ts:114]: But type wise I have to return string | undefined which violates ts strict. That it is how it was before I added the new use case

Comment (rebelchris) [reply src/common/cloudinary.ts:114]: Ah I see, not the biggest fan of overloads in JS world, but guess it's ok if it works 😂

**Review by rebelchris — APPROVED** [review]
Looks ok

**Pull request merged**